Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+                indoorsLocationDataSource = IndoorsLocationDataSource(definition: indoorPositioningDefinition)
+            } else {
+                indoorsLocationDataSource = try await createIndoorLocationDataSourceFromTables(map: map)
+            }
+            // This ensures that the details of the inside of the building, like room layouts are displayed.
+            for featLayer in map.operationalLayers {
+                if featLayer.name == "Transitions" || featLayer.name == "Details" {
+                    featLayer.isVisible = true
+                }
+            }
+            // The indoorsLocationDataSource should always be there. Since the createIndoorLocationDataSourceFromTables returns
+            // an optional value it cannot be guaranteed. Best option if you get to this point without a datasource
+            // is to return (ideally an error would have been thrown before this point and the flow broken.)
+            guard let dataSource = indoorsLocationDataSource else { return }
+            locationDisplay.dataSource = dataSource
+            locationDisplay.autoPanMode = .compassNavigation
+            // Start the location display to zoom to the user's current location.
+            try await locationDisplay.dataSource.start()
+        }
+        
+        /// Creates an indoor location datasource from the maps tables if there is no indoors definition.
+        /// - Parameter map: The map which contains the tables from which the data source is constructed.
+        /// - Returns: Returns a configured IndoorsLocationDataSource created from the IPS position table.
+        private func createIndoorLocationDataSourceFromTables(map: Map) async throws -> IndoorsLocationDataSource? {
+            // Gets the positioning table from the map.
+            guard let positioningTable = map.tables.first(where: { $0.displayName == "IPS_Positioning" }) else { return nil }
+            // Creates and configures the query parameters.
+            let queryParameters = QueryParameters()
+            queryParameters.maxFeatures = 1
+            queryParameters.whereClause = "1 = 1"
+            // Queries positioning table to get the positioning ID.
+            let queryResult = try await positioningTable.queryFeatures(using: queryParameters)
+            guard let feature = queryResult.features().makeIterator().next() else { return nil }
+            let serviceFeatureTable = positioningTable as! ServiceFeatureTable
+            let positioningID = feature.attributes[serviceFeatureTable.globalIDField] as? UUID
+            
+            // Gets the pathways layer (optional for creating the IndoorsLocationDataSource).
+            let pathwaysLayer = map.operationalLayers.first(where: { $0.name == "Pathways" }) as! FeatureLayer
+            // Gets the levels layer (optional for creating the IndoorsLocationDataSource).
+            let levelsLayer = map.operationalLayers.first(where: { $0.name == "Levels" }) as! FeatureLayer
```

Comment:
Here you're assuming that these layers will always be there by force unwrapping them, but you didn't do that for the positioning table. How come?

Reply:
I had it previously with the if let, but I looked at the how we have it in the guide online and wanted to keep the code as close to what was posted to users on the web as possible. I would probably not force unwrap them if it was purely my implementation, because its possible the data could not be there and then you'd have the app store version crash always without much explanation.

Reply:
If we're genuinely concerned that the data may change, then we should be consistently safe and not force unwrap any of the data. Check with your team, but I think the likelihood of the data changing is so low that it would be fine to force unwrap, provided that it is guaranteed that the map has already been loaded.

Reply:
Okay sounds good!